src/mlpack/methods/ann/models/yolov3/yolov3_layer_impl.hpp

```diff
@@ -0,0 +1,254 @@
+
+  // Input dimensions: gridSize
+  MatType offset = arma::regspace<MatType>(0, gridSize - 1);
+
+#if ARMA_VERSION_MAJOR < 15
+  // If arma::repcube is not available
+  CubeType anchorsWBS(anchorsW.n_rows, anchorsW.n_cols, batchSize);
+  CubeType anchorsHBS(anchorsW.n_rows, anchorsW.n_cols, batchSize);
+  CubeType xOffset(grid, predictionsPerCell, batchSize);
+
+  arma::Col<Type> offsetT =
+    arma::vectorise(arma::repmat(offset.t(), gridSize, 1));
+  CubeType yOffset(grid, predictionsPerCell, batchSize);
+  for (size_t i = 0; i < batchSize; i++)
+  {
+    anchorsWBS.slice(i) = anchorsW;
+    anchorsWBS.slice(i) = anchorsH;
+    xOffset.slice(i) = arma::repmat(offset, gridSize, predictionsPerCell);
+    yOffset.slice(i) = arma::repmat(offsetT, 1, predictionsPerCell);
+  }
+#else
+  CubeType anchorsWBS = arma::repcube(anchorsW, 1, 1, batchSize);
+  CubeType anchorsHBS = arma::repcube(anchorsH, 1, 1, batchSize);
+  CubeType xOffset = arma::repcube(offset, gridSize,
+    predictionsPerCell, batchSize);
+
+  CubeType yOffset = arma::repcube(
+    arma::vectorise(arma::repmat(offset.t(), gridSize, 1)),
+    1, predictionsPerCell, batchSize);
+#endif
+
+  // TODO: add if (this->training). Add check for different batchSize.
+  const size_t cols = predictionsPerCell - 1;
+  // x
+  outputCube.tube(0, 0, grid - 1, cols) =
+    (xOffset + 1 / (1 + arma::exp(-inputCube.tube(0, 0, grid - 1, cols))))
+    * stride;
+
+  // y
+  outputCube.tube(grid, 0, grid * 2 - 1, cols) = (yOffset + 1 /
+    (1 + arma::exp(-inputCube.tube(grid, 0, grid * 2 - 1, cols)))) * stride;
+
+  // w
+  outputCube.tube(grid * 2, 0, grid * 3 - 1, cols) =
+    anchorsWBS % arma::exp(inputCube.tube(grid * 2, 0, grid * 3 - 1, cols));
+
+  // h
+  outputCube.tube(grid * 3, 0, grid * 4 - 1, cols) =
+    anchorsHBS % arma::exp(inputCube.tube(grid * 3, 0, grid * 4 - 1, cols));
+
+  // apply logistic sigmoid to objectness and classification logits.
+  outputCube.tube(grid * 4, 0, outputCube.n_rows - 1, cols) = 1. /
+    (1 + arma::exp(-inputCube.tube(grid * 4, 0, inputCube.n_rows - 1, cols)));
+
+  // Reshape, for each batch item.
+  for (size_t i = 0; i < reshapedCube.n_slices; i++)
+  {
+    reshapedCube.slice(i) = arma::reshape(
+      arma::reshape(
+        outputCube.slice(i), grid, numAttributes * predictionsPerCell).t(),
+        numAttributes, predictionsPerCell * grid);
+  }
+}
+
+template <typename MatType>
+void YOLOv3Layer<MatType>::Backward(
+    const MatType& input,
+    const MatType& output,
+    const MatType& gy,
+    MatType& g)
+{
+  throw std::runtime_error("YOLOv3::Backward() not implemented.");
+}
+
+template <typename MatType>
+template <typename Archive>
+void YOLOv3Layer<MatType>::serialize(Archive& ar, const uint32_t /* version */)
+{
+  ar(cereal::base_class<Layer<MatType>>(this));
+  ar(CEREAL_NVP(imgSize));
+  ar(CEREAL_NVP(numAttributes));
+  ar(CEREAL_NVP(gridSize));
+  ar(CEREAL_NVP(grid));
+  ar(CEREAL_NVP(anchorsW));
+  ar(CEREAL_NVP(anchorsH));
```

Comment:
Is it possible we could avoid the need for these and instead inline their definitions into where they're used?  If not, you could at least avoid serializing them and just clear/regenerate them on a load and that should help keep the serialized model smaller.